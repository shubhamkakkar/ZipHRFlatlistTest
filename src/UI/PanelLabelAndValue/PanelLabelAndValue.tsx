import React from 'react';
import { View, Text } from 'react-native';
import styles from './styleSheetPanelLabelAndValue';

type TPanelLabelAndValue = {
  title: string;
  value: number | string;
};

export default function PanelLabelAndValue({ title, value }: TPanelLabelAndValue) {
  return (
    <View style={[styles.container]}>
      <Text style={[styles.panelTextLabel]}>{title}</Text>
      <Text style={[styles.panelTextValue]}>{value && value.toString().trim()}</Text>
    </View>
  );
}
