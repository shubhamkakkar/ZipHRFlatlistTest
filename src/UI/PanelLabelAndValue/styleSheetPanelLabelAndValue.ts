import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    marginBottom: 15,
    marginHorizontal: 5,
    flex: 1,
    justifyContent: 'center',
  },
  panelTextLabel: {
    color: '#777777',
    fontSize: 14,
    lineHeight: 19,
  },
  panelTextValue: {
    fontSize: 16,
    lineHeight: 19,
    marginTop: 5,
    flexWrap: 'wrap',
    flexDirection: 'row',
  },
});

export default styles;
