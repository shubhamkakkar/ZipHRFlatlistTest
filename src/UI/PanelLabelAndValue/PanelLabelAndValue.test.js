import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PanelLabelAndValue from './PanelLabelAndValue';

it('PanelLabelAndValue matches snapshot - props: {{ title: "", value: "" }}', () => {
  const tree = renderer.create(<PanelLabelAndValue {...{ title: '', value: '' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('PanelLabelAndValue matches snapshot - props: {{ title: "", value: 0 }}', () => {
  const tree = renderer.create(<PanelLabelAndValue {...{ title: '', value: 0 }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('PanelLabelAndValue matches snapshot - props: {{ title: "title", value: "" }}', () => {
  const tree = renderer.create(<PanelLabelAndValue {...{ title: 'title', value: '' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('PanelLabelAndValue matches snapshot - props: {{ title: "title", value: 0 }}', () => {
  const tree = renderer.create(<PanelLabelAndValue {...{ title: 'title', value: 0 }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
