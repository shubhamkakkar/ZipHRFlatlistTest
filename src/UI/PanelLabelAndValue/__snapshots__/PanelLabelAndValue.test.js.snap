// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelLabelAndValue matches snapshot - props: {{ title: "", value: "" }} 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 15,
        "marginHorizontal": 5,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#777777",
          "fontSize": 14,
          "lineHeight": 19,
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "fontSize": 16,
          "lineHeight": 19,
          "marginTop": 5,
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`PanelLabelAndValue matches snapshot - props: {{ title: "", value: 0 }} 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 15,
        "marginHorizontal": 5,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#777777",
          "fontSize": 14,
          "lineHeight": 19,
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "fontSize": 16,
          "lineHeight": 19,
          "marginTop": 5,
        },
      ]
    }
  >
    0
  </Text>
</View>
`;

exports[`PanelLabelAndValue matches snapshot - props: {{ title: "title", value: "" }} 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 15,
        "marginHorizontal": 5,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#777777",
          "fontSize": 14,
          "lineHeight": 19,
        },
      ]
    }
  >
    title
  </Text>
  <Text
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "fontSize": 16,
          "lineHeight": 19,
          "marginTop": 5,
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`PanelLabelAndValue matches snapshot - props: {{ title: "title", value: 0 }} 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 15,
        "marginHorizontal": 5,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#777777",
          "fontSize": 14,
          "lineHeight": 19,
        },
      ]
    }
  >
    title
  </Text>
  <Text
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "fontSize": 16,
          "lineHeight": 19,
          "marginTop": 5,
        },
      ]
    }
  >
    0
  </Text>
</View>
`;
